scripts/apps/authoring-react/widget-persistance-hoc.tsx

```diff
@@ -0,0 +1,50 @@
+import {closedThroughAction} from 'apps/authoring/widgets/widgets';
+import React, {RefObject} from 'react';
+import {IArticleSideWidgetComponentType} from 'superdesk-api';
+
+export let widgetState = {};
+
+interface IProps {
+    children: (
+        widgetRef: RefObject<React.PureComponent<IArticleSideWidgetComponentType>>,
+    ) => JSX.Element | Array<JSX.Element>;
+    sideWidgetId: string;
+    pinned: boolean;
+}
+
+interface IState {
+    mounted: boolean;
+}
+
+export class WidgetStatePersistanceHOC extends React.PureComponent<IProps, IState> {
+    private widgetRef: RefObject<React.PureComponent<IArticleSideWidgetComponentType, any>>;
+
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            mounted: false,
+        };
+        this.widgetRef = React.createRef<React.PureComponent<IArticleSideWidgetComponentType, any>>();
+    }
+
+    componentDidMount(): void {
+        this.setState({mounted: true});
+    }
+
+    componentWillUnmount(): void {
+        if (this.widgetRef?.current != null) {
+            widgetState[this.props.sideWidgetId] = this.widgetRef.current.state;
+        }
+
+        // Reset widgetState if widget was closed through a function, or
+        // if it wasn't pinned and got closed from re-rendering
+        if (closedThroughAction.closed === true) {
+            delete widgetState[this.props.sideWidgetId];
+        }
+    }
+
+    render(): React.ReactNode {
+        return this.state.mounted ? this.props.children(this.widgetRef) : <></>;
```

Comment:
use `null` if your intention is not to render anything. Use a fragment only if you need to return multiple sibling elements without wrapping it.